libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/ScopedSpan.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing;
+
+import org.opensearch.telemetry.tracing.noop.NoopScopedSpan;
+
+/**
+ * An auto-closeable that represents scoped span.
+ * It provides interface for all the span operations.
+ */
+public interface ScopedSpan extends AutoCloseable {
+    /**
+     * No-op Scope implementation
+     */
+    ScopedSpan NO_OP = new NoopScopedSpan();
+
+    /**
+     * Adds string attribute to the {@link Span}.
+     *
+     * @param key   attribute key
+     * @param value attribute value
+     */
+    void addSpanAttribute(String key, String value);
```

Comment:
```suggestion
    void addAttribute(String key, String value);
```

I think we could drop `xxxSpanXxx`  from methods, the class itself says it is `XxxSpan`